sql/src/main/java/io/crate/operation/projectors/SortingProjector.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.operation.projectors;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Ordering;
+import com.google.common.collect.Sets;
+import io.crate.core.collections.CollectionBucket;
+import io.crate.core.collections.Row;
+import io.crate.operation.Input;
+import io.crate.operation.collect.CollectExpression;
+
+import java.util.*;
+
+/**
+ * Sort rows by ordering criteria and process given offset before emitting.
+ *
+ * Compared to {@link SortingTopNProjector} this projector does not support limiting,
+ * while the {@link SortingTopNProjector} does not work WITHOUT a limit.
+ */
+class SortingProjector extends AbstractProjector {
+
+    protected final Collection<? extends Input<?>> inputs;
+    protected final Iterable<? extends CollectExpression<Row, ?>> collectExpressions;
+    protected Set<Requirement> requirements;
+
+    private final Ordering<Object[]> ordering;
+    private List<Object[]> rows = new ArrayList<>();
```

Comment:
Couldn't you pass the rows.subList into createRowEmitter so that the re-assignment isn't necessary?